packages/eas-cli/src/utils/profiles.ts

```diff
@@ -65,7 +65,8 @@ async function readProfileAsync<T extends ProfileType>({
       profileName
     );
 
-    maybePrintBuildProfileDeprecationWarnings(buildProfile, profileName);
+    const rawEasJson = await easJsonAccessor.readRawJsonAsync();
```

Comment:
Consider passing easJsonAccessor all the way and running readRawJsonAsync when you need the value, to avoid passing any through a bunch of functions.